### PR TITLE
chore(webchat): compilable

### PR DESCRIPTION
**Pull request by samuelmasse**
Makes webchat compile in the messaging repo. The codebase changes to strict typescript, with no implicit any. The changes I do here aren't very smart they're mainly to get the code to actually compile.

Closes DEV-1884

### CODE REVIEW

File: packages/webchat/src/components/Composer.tsx [hunk 1]
```diff
@@ -3,61 +3,61 @@ import { inject, observer } from 'mobx-react'
 import React from 'react'
 import { FormattedMessage, InjectedIntlProps, injectIntl } from 'react-intl'
 
-import ToolTip from '../../../../../../packages/ui-shared-lite/ToolTip'
+// import ToolTip from '../../../../../../packages/ui-shared-lite/ToolTip'
 import { RootStore, StoreDef } from '../store'
```
Comment (samuelmasse) [inline packages/webchat/src/components/Composer.tsx:7]: We can't import the tooltip from ui-shared-lite here.

File: packages/webchat/src/components/ContextMenu.tsx [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
     <Menu>
-      {props.customActions.map(action => {
+      {props.customActions.map((action: any) => {
         return (
-          <MenuItem key={action.id} text={action.label} onClick={action.onClick.bind(this, action.id, props.element)} />
+          <MenuItem
+            key={action.id}
+            text={action.label}
+            // TODO: undefined is supposed to be "this". Does that make sense??
+            onClick={action.onClick.bind(undefined, action.id, props.element)}
```
Comment (samuelmasse) [inline packages/webchat/src/components/ContextMenu.tsx:13]: Before we had `onClick.bind(this, ...)` but typescript tells us this makes no sense and produces a typing error.

File: packages/webchat/src/components/Header.tsx [hunk 3]
```diff
@@ -2,8 +2,9 @@ import { observe } from 'mobx'
 import { inject, observer } from 'mobx-react'
 import React from 'react'
 
-import confirmDialog from '../../../../../../packages/ui-shared-lite/ConfirmDialog'
-import MoreOptions from '../../../../../../packages/ui-shared-lite/MoreOptions'
+// TODO: put this back
+// import confirmDialog from '../../../../../../packages/ui-shared-lite/ConfirmDialog'
+// import MoreOptions from '../../../../../../packages/ui-shared-lite/MoreOptions'
```
Comment (samuelmasse) [inline packages/webchat/src/components/Header.tsx:7]: We can't import this stuff from ui-shared-lite aswell. Do we copy paste this then?

File: packages/webchat/src/components/Header.tsx [hunk 4]
```diff
@@ -280,7 +284,7 @@ class Header extends React.Component<HeaderProps> {
           <span className="bpw-emulator-header-tab">Emulator</span>
           <div>
             <span className="bpw-emulator-buttons">{this.props.showResetButton && this.renderResetButton()}</span>
-            <MoreOptions show={this.state.showingOption} onToggle={this.setShowingOption} items={optionsItems} />
+            {/* <MoreOptions show={this.state.showingOption} onToggle={this.setShowingOption} items={optionsItems} /> */}
```
Comment (samuelmasse) [inline packages/webchat/src/components/Header.tsx:287]: Commented out ui-shared-lite component

File: packages/webchat/src/components/Header.tsx [hunk 5]
```diff
@@ -77,23 +78,26 @@ class Header extends React.Component<HeaderProps> {
...
+          declineLabel: this.props.intl!.formatMessage({
             id: 'header.deleteConversationNo'
           })
         }
       )
     ) {
-      await this.props.deleteConversation()
+      await this.props.deleteConversation!()
     }
+    */
```
Comment (samuelmasse) [inline packages/webchat/src/components/Header.tsx:100]: Commented out ui-shared-lite component

File: packages/webchat/src/components/VoiceRecorder.tsx [hunk 6]
```diff
@@ -2,8 +2,9 @@ import cx from 'classnames'
 import mime from 'mime/lite'
 import React, { FC, useCallback, useState, useEffect, useRef, Fragment } from 'react'
 
-import Cancel from '../../../../../../packages/ui-shared-lite/Icons/Cancel'
-import Microphone from '../../../../../../packages/ui-shared-lite/Icons/Microphone'
+// TODO: put this back
+// import Cancel from '../../../../../../packages/ui-shared-lite/Icons/Cancel'
+// import Microphone from '../../../../../../packages/ui-shared-lite/Icons/Microphone'
```
Comment (samuelmasse) [inline packages/webchat/src/components/VoiceRecorder.tsx:7]: Other ui-shared-lite components we can't import

File: packages/webchat/src/components/messages/MessageGroup.tsx [hunk 7]
```diff
@@ -3,7 +3,8 @@ import sortBy from 'lodash/sortBy'
 import { inject } from 'mobx-react'
 import React from 'react'
 
-import { renderPayload } from '../../../../../../../packages/ui-shared-lite/Payloads'
+// TODO: what to do with this?
+// import { renderPayload } from '../../../../../../../packages/ui-shared-lite/Payloads'
```
Comment (samuelmasse) [inline packages/webchat/src/components/messages/MessageGroup.tsx:7]: Fix: we'll juste use the components package from Charles.

File: packages/webchat/src/components/messages/MessageGroup.tsx [hunk 8]
```diff
@@ -90,12 +91,16 @@ class MessageGroup extends React.Component<Props> {
...
+              const payload = this.convertPayloadFromOldFormat(message)
+
+              /*
+              // TODO: what to do with this?
               if (payload?.wrapped) {
                 payload = { ...payload, wrapped: renderPayload(payload.wrapped) }
               } else {
                 payload = renderPayload(payload)
               }
+              */
```
Comment (samuelmasse) [inline packages/webchat/src/components/messages/MessageGroup.tsx:103]: Use the components package

**Review by samuelmasse — COMMENTED** [review]
@laurentlp Should be good for a first review. We'll have to see what we do with the ui-shared-lite components from the main repo. Either we copy them to this package or we publish the ui-shared-lite package and add a dependency to it (not great). @charlescatta did you have any dependencies that you had to resolve to ui-shared-lite components when making you package?

**Review by laurentlp — APPROVED** [review]

**Comment by charlescatta** [comment]
@samuelmasse non I had no ui-shared* dependencies

File: packages/components/src/typings.d.ts [hunk 9]
```diff
@@ -37,6 +37,7 @@ export interface MessageConfig {
   onSendData: (data: any) => Promise<void>
   onFileUpload: FileUploadHandler
   onMessageClicked: (messageId?: uuid) => void
+  // @ts-ignore
```
[the remark this answers is not in the document]
Comment (samuelmasse) [reply packages/components/src/typings.d.ts:40]: @charlescatta this line doesn't compile in typescript so I added a ts-ignore

**Pull request merged**